crates/polars-stream/src/nodes/zip.rs

```diff
@@ -333,7 +347,8 @@ impl ComputeNode for ZipNode {
                 for input_head in &mut self.input_heads {
                     out.push(input_head.consume_broadcast());
                 }
-                let out_df = concat_df_horizontal(&out, false)?;
+                let strict_concat = matches!(self.extend_behavior, ExtendBehavior::Raise);
+                let out_df = concat_df_horizontal(&out, false, strict_concat)?;
```

Comment:
Same here.

Reply:
If we are in the branch all input heads are broadcast inputs, meaning they have length 1.